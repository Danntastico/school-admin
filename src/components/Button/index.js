import React from "react";
import { Btn } from "./styles";
import PropTypes from "prop-types";

export const Button = ({ onClick, label = "Button", isDisabled }) => {
  return (
    <Btn onClick={onClick} disabled={isDisabled}>
      {label}
    </Btn>
  );
};
