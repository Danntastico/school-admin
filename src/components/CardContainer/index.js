import React from 'react';
import { Container } from './styles';
export const CardContainer = ({ children }) => {
  return <Container>{children}</Container>;
};
