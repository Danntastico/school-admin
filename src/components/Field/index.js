import React from "react";
import { Input, Label, FieldContainer } from "./styles";
export const Field = ({
  disabled = false,
  fieldType,
  handleInputChange,
  inputType,
  label,
  name,
  placeholder,
  value,
  children,
}) => {
  const fieldRender = () => {
    switch (fieldType) {
      case "textarea":
        return (
          <textarea
            autoComplete="off"
            disabled={disabled}
            name={name}
            id={name}
            onChange={handleInputChange}
            value={value}
          />
        );
      case "select":
        return (
          <select name={name} value={value} onChange={handleInputChange}>
            {children}
          </select>
        );
      default:
        return (
          <Input
            autoComplete="off"
            disabled={disabled}
            id={name}
            name={name}
            onChange={handleInputChange}
            placeholder={placeholder}
            type={inputType}
            value={value}
            required={true}
          />
        );
    }
  };
  return (
    <FieldContainer>
      <Label htmlFor={name}> {label}</Label>
      {fieldRender()}
    </FieldContainer>
  );
};
