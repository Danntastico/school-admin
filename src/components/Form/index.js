import React from 'react';
import { FormHeader, FormTitle, FormBody, FormFooter } from './styles';
export const Form = ({ children, title, onSubmit }) => {
  return (
    <form onSubmit={onSubmit}>
      <FormHeader>
        <FormTitle> {title} </FormTitle>
      </FormHeader>
      <FormBody>{children[0]}</FormBody>
      <FormFooter>{children[1]}</FormFooter>
    </form>
  );
};
