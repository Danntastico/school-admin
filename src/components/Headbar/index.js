import React from 'react';

export const Headbar = ({ title }) => {
  return (
    <header>
      <h3>title</h3>
    </header>
  );
};
