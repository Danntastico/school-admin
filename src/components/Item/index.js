import React from "react";
import { ItemList } from "./styles";

export const Item = ({ children, bgColor, txtColor, to }) => (
  <ItemList to={to} bgcolor={bgColor} txtcolor={txtColor}>
    {children}
  </ItemList>
);
