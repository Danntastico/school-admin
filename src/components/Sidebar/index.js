import React from 'react';
import { MdHome, MdWork, MdLocalLibrary, MdLibraryBooks } from 'react-icons/md';
import {
  HOME_PATH,
  TEACHER_PATH,
  STUDENT_PATH,
  SUBJECT_PATH,
} from '../../utils/constants';

import {
  SidebarContainer,
  NavList,
  ItemList,
  SidebarHeader,
  Title,
  SidebarFooter,
} from './styles';

export const Sidebar = () => {
  return (
    <SidebarContainer>
      <SidebarHeader>
        <Title>SCHOOL Admin</Title>
      </SidebarHeader>
      <NavList>
        <ItemList to={HOME_PATH}>
          <MdHome />
          Home
        </ItemList>
        <ItemList to={TEACHER_PATH}>
          <MdWork />
          Teachers
        </ItemList>
        <ItemList to={STUDENT_PATH}>
          <MdLocalLibrary />
          Students
        </ItemList>
        <ItemList to={SUBJECT_PATH}>
          <MdLibraryBooks />
          Subjects
        </ItemList>
      </NavList>
      <SidebarFooter>
        <button>GENERATE REPORT</button>
      </SidebarFooter>
    </SidebarContainer>
  );
};
