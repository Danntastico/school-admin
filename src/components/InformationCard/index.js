import React from "react";
import { MdEdit } from "react-icons/md";
import { InfoCard, InfoHead, InfoBody, InfoTitle } from "./styles";
import { EditBtn } from "../Form/styles";
export const InformationCard = ({
  children,
  handleClickOnEdit,
  title,
  isEditable,
}) => {
  return (
    <InfoCard>
      <InfoHead>
        <InfoTitle>{title}</InfoTitle>
        {isEditable && (
          <EditBtn onClick={handleClickOnEdit}>
            <MdEdit />
          </EditBtn>
        )}
      </InfoHead>
      <InfoBody>{children}</InfoBody>
    </InfoCard>
  );
};
