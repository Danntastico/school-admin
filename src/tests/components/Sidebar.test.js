import React from 'react';
import { Sidebar } from '../../components/Sidebar';
import { shallow } from 'enzyme';

import '@testing-library/jest-dom';

describe('Unit Test for <Sidebar />', () => {
  const wrapper = shallow(<Sidebar />);

  test('Should show sidebar correctly (Shallow copy)', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('Should render 4 different links', () => {
    const navList = wrapper.prop('children')[1].props.children;
    expect(navList.length).toBe(4);
  });
});
