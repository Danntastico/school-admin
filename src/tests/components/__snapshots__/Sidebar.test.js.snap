// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Unit Test for <Sidebar /> Should show sidebar correctly (Shallow copy) 1`] = `
<styled.nav>
  <styled.div>
    <styled.h2>
      SCHOOL Admin
    </styled.h2>
  </styled.div>
  <styled.div>
    <Styled(Link)
      to="/"
    >
      <MdHome />
      Home
    </Styled(Link)>
    <Styled(Link)
      to="teachers/*"
    >
      <MdWork />
      Teachers
    </Styled(Link)>
    <Styled(Link)
      to="students/*"
    >
      <MdLocalLibrary />
      Students
    </Styled(Link)>
    <Styled(Link)
      to="subjects/*"
    >
      <MdLibraryBooks />
      Subjects
    </Styled(Link)>
  </styled.div>
  <styled.div>
    <button>
      GENERATE REPORT
    </button>
  </styled.div>
</styled.nav>
`;
